src/mlstacks/analytics/client.py

```diff
@@ -161,15 +162,23 @@ def track_event(
         metadata: Dict of metadata to track.
 
     Returns:
-        True if event is sent successfully, False is not.
+        True if event is sent successfully, False otherwise.
     """
     if metadata is None:
         metadata = {}
 
     metadata.setdefault("event_success", True)
 
     with MLStacksAnalyticsContext() as analytics_context:
-        return bool(analytics_context.track(event=event, properties=metadata))
+        try:
```

Comment:
This is where it gets tricky a bit. 

The `MLStacksAnalyticsContext` is implemented as a context manager. Any `Exception` that would happen during the execution of things within the scope of this context manager will be handled by its `__exit__` method. This way, we ensure that actual execution can not fail, even if something goes wrong in the analytics. So this `try-except` is already covered by it.

You may have seen some error messages already though. The reason is, by default, the segment analytics python package is using something called a [`Consumer`](https://github.com/segmentio/analytics-python/blob/1.4.0/analytics/consumer.py), which creates separate threads to upload and send out the events. Due to its nature, any calls happening within this thread are out of the scope of our context manager. However, if something goes wrong, they handle it the same way with a `try-catch` and give out an error log right [here](https://github.com/segmentio/analytics-python/blob/bf7168d7cb9bed5b1599928bca4d83fd468f31ce/analytics/consumer.py#L71), that you may have seen already.

I see two solutions if you would like to get rid of the error logs: 

1. You can disable the usage of this consumer paradigm by setting the `analytics.sync_mode` to True. This way, the events will be sent out by the main thread, and the `MLStacksAnalyticsContext` will do all the error handling. However, in the case of an unresponsive server, this will block the main thread for `analytics.max_retries+1` * `analytics.timeout` seconds, so it is not very ideal.
2. You can try to disable the logger for the segment analytics package and implement a custom `analytics.on_error` method to handle the same error message as a debug message.

Personally, I would recommend the second solution.

Reply:
Thank you for the thorough explanation about the `MLStacksAnalyticsContext` and Segment's consumer threads behavior. These issues were noted in the original GitHub ticket, which led me to initially consider a threading-based solution to address the main problem. However, as you mentioned, they are outside the scope of mlstacks' analytics client. 

Given the nuances you've outlined,  I agree that the second solution seems more appropriate. I'll work on implementing the `analytics.on_error` method, with non-disruptive error handling.

Reply:
I've implemented the custom `on_error` handler as recommended, including corresponding unit tests to verify its functionality. If there are any further adjustments or tests you think should be included, please let me know.